keel-echo/src/main/kotlin/com/netflix/spinnaker/keel/echo/ManualJudgementNotifier.kt

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.keel.echo
+
+import com.netflix.spinnaker.keel.api.ConstraintStatus
+import com.netflix.spinnaker.keel.api.ManualJudgementConstraint
+import com.netflix.spinnaker.keel.api.application
+import com.netflix.spinnaker.keel.echo.model.EchoNotification
+import com.netflix.spinnaker.keel.events.ConstraintStateChanged
+import kotlinx.coroutines.runBlocking
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+/**
+ * Listens for [ConstraintStateChanged] events where the constraint is a [ManualJudgementConstraint] and sends
+ * out notifications so that users can take action.
+ */
+class ManualJudgementNotifier(
+  private val echoService: EchoService
+) {
+
+  @EventListener(ConstraintStateChanged::class)
+  fun constraintStateChanged(event: ConstraintStateChanged) {
+    // if this is the first time the constraint was evaluated, send a notification
+    // so the user can react via other interfaces outside the UI (e.g. e-mail, Slack)
+    if (event.constraint is ManualJudgementConstraint &&
+      event.previousState == null &&
+      event.currentState.status == ConstraintStatus.PENDING) {
+      event.environment.notifications.map {
+        // TODO: run in parallel
+        runBlocking {
+          echoService.sendNotification(
```

Comment:
This is a lot of configuration for the notification wrapped up in one function. Do you think it's worth pulling it out into a separate function? Something like `fun Notification.toEchoNotification(event: ConstraintStateChanged)` or something? 

I always like to split up the creation of the object when the initialization has logic in it.

Reply:
Sounds good. Will make that change.